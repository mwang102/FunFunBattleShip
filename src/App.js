import React, { Component } from 'react';
import styled from 'styled-components';
import BattleShipBoard from './components/BattleShipBoard'
import '../node_modules/animate.css/animate.css'
import InfoBoard from './components/InfoBoard'
import Header from './components/Header'
import { handleInfoBoard, resetInfoBoard} from './components/helper-functions/app-state-functions'


class App extends Component {
  constructor(){
    super()

    this.state = {
      board:[
        [[5,false],[0,false],[0,false],[0,false]],
        [[5,false],[0,false],[0,false],[0,false]],
        [[5,false],[2,false],[2,false],[0,false]],
        [[0,false],[0,false],[0,false],[0,false]]
      ],
      turn:0,
      hits:0,
      miss:0,
      sunkBoats:0,
      alreadyHit:0,
      numberHitsLeft:5,
    }

    this.handlePick = this.handlePick.bind(this)
  }

  handlePick(event){ 
    const boatHit = event.data[1] === true ? 0 : event.data[0] > 0 ? 1 : 0,
          correctState = this.state.numberHitsLeft - boatHit === 0 ? resetInfoBoard() : handleInfoBoard(this.state,event)

    this.setState(correctState)
  }


  render() {
    const {  
      board, 
      turn,
      hits,
      miss,
      sunkBoats,
      alreadyHit,
      numberHitsLeft,
    } = this.state

    return (
      <AppContainer>
        <Header/>
        <h1>
          To get started, click a square!
        </h1>
        <MainSection>
            <InfoBoard numberHitsLeft={numberHitsLeft} alreadyHit={alreadyHit} turn={turn} hits={hits} miss={miss} sunkBoats={sunkBoats}/>
            <BattleShipBoard handlePick={this.handlePick} board={board}/>
        </MainSection>
      </AppContainer>
    );
  }
}

const MainSection = styled.section`
    display:flex;
    margin:0 10rem;
    flex:1;


`

const AppContainer = styled.div`
    display:flex;
    flex-direction:column;
    text-align:center;
    font-size: 16px;
    font-family: Helvetica, Arial, sans-serif;
    margin:0;
    padding:0;
    min-height:100vh;
`

export default App;
