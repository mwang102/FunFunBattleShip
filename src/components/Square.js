import React from 'react'
import styled from "styled-components";
import PropTypes from 'prop-types'

    /* 2 flipped and hit
       1 not flipped yet and hit 
       0 not flipped and nothing under
      -1 flipped and nothing under 
    */

export default class Square extends React.Component{
  static propTypes = {
    rowNumber:PropTypes.number.isRequired,
    columnNumber:PropTypes.number.isRequired,
    data:PropTypes.array.isRequired
  }

  render(){
    const { data, rowNumber, columnNumber } = this.props
    const currentSquare = {
        row:rowNumber, 
        column:columnNumber, 
        data:data
        },
        squareBackground = data[0] > 0 && data[1] === true ? 'rgb(' + (80 + (data[0] * 50)) + ',0,0)' : 
                           data[0] ===  -1 ? 'SteelBlue' : 'BlanchedAlmond',
        hoverBackground = data.squareValue === 2 || data.squareValue === -1 ? '' : 'gold'

    return(
      <SquareContainer hoverBackground={hoverBackground} backGround = {squareBackground} squareValue={currentSquare.data}
        onClick={()=> this.props.handlePick(currentSquare)}>
      </SquareContainer>
    )
  }
}

const SquareContainer = styled.div`
    display:flex;
    justify-content:center;
    align-items:center;
    margin: 0px 5px;
    flex: 1 0 auto;
    padding: 0px 1.0875rem 1.45rem;
    box-shadow: 0 1px 5px 0 rgba(0,0,0,0.5);
    font-size:30px;
    background-repeat:no-repeat;
    background-position: center; 
    background:${props => props.backGround };
    cursor:${props => props.squareValue[1] ? '' : 'pointer'}};
    -webkit-animation:${props => props.squareValue[1] ? 'flipInX 1s' : ''}};
    :hover{
      background:${props => props.hoverBackground }
`;
