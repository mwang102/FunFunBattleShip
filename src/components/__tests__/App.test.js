import React from 'react';
import ReactDOM from 'react-dom';
import App from '../../App';
import Square from '../Square'
import { shallow, mount } from 'enzyme'
import renderer from 'react-test-renderer'

describe('App', ()=>{
	it('renders without crashing', () => {
	  const div = document.createElement('div');
	  ReactDOM.render(<App />, div);
	  ReactDOM.unmountComponentAtNode(div);
	});


	it('checks if IntoBoard gets updated correctly when only a hit occurs', ()=>{
		const wrapper = mount(<App/>)
		wrapper.setState({ board:[
	        [[5,false],[0,false],[0,false],[0,false]],
	        [[5,false],[0,false],[0,false],[0,false]],
	        [[5,false],[2,false],[2,false],[0,false]],
	        [[0,false],[0,false],[0,false],[0,false]]
	      ]
		})
		const firstSquare = wrapper.find('Square').at(0)
		firstSquare.simulate('click')

		const infoBoardTableTurnData = wrapper.find('InfoBoard__TableData').at(1).render().text()
		expect(infoBoardTableTurnData).toBe('1')

		const infoBoardTableHitData = wrapper.find('InfoBoard__TableData').at(3).render().text()
		expect(infoBoardTableHitData).toBe('1')

		const infoBoardTableNumberHitsLeft = wrapper.find('InfoBoard__TableData').at(11).render().text()
		expect(infoBoardTableNumberHitsLeft).toBe('4')
	})

	it('checks if IntoBoard gets updated correctly when a miss occurs', ()=>{
		const wrapper = mount(<App/>)
		wrapper.setState({ board:[
	        [[5,false],[0,false],[0,false],[0,false]],
	        [[5,false],[0,false],[0,false],[0,false]],
	        [[5,false],[2,false],[2,false],[0,false]],
	        [[0,false],[0,false],[0,false],[0,false]]
	      ]
		})
		const secondSquare = wrapper.find('Square').at(1)
		secondSquare.simulate('click')

		const infoBoardTableTurnData = wrapper.find('InfoBoard__TableData').at(1).render().text()
		expect(infoBoardTableTurnData).toBe('1')

		const infoBoardTableMissData = wrapper.find('InfoBoard__TableData').at(5).render().text()
		expect(infoBoardTableMissData).toBe('1')
	})

	it('checks if IntoBoard gets updated correctly when we sink a ship', ()=>{
		const wrapper = mount(<App/>)
		wrapper.setState({ board:[
	        [[5,false],[0,false],[0,false],[0,false]],
	        [[5,false],[0,false],[0,false],[0,false]],
	        [[5,false],[2,false],[2,false],[0,false]],
	        [[0,false],[0,false],[0,false],[0,false]]
	      ]
		})
		const firstSquare = wrapper.find('Square').at(0)
		const fourthSquare = wrapper.find('Square').at(4)
		const eighthSquare = wrapper.find('Square').at(8)
		firstSquare.simulate('click')
		fourthSquare.simulate('click')
		eighthSquare.simulate('click')

		const infoBoardTableTurnData = wrapper.find('InfoBoard__TableData').at(1).render().text()
		expect(infoBoardTableTurnData).toBe('3')

		const infoBoardTableHitData = wrapper.find('InfoBoard__TableData').at(3).render().text()
		expect(infoBoardTableHitData).toBe('3')

		const infoBoardTableNumberHitsLeft = wrapper.find('InfoBoard__TableData').at(11).render().text()
		expect(infoBoardTableNumberHitsLeft).toBe('2')

		const infoBoardTableNumberSunk = wrapper.find('InfoBoard__TableData').at(7).render().text()
		expect(infoBoardTableNumberSunk).toBe('1')
	})

	it('checks if IntoBoard gets updated correctly when we click a spot already targeted', ()=>{
		const wrapper = mount(<App/>)
		wrapper.setState({ board:[
	        [[5,true],[0,false],[0,false],[0,false]],
	        [[5,false],[0,false],[0,false],[0,false]],
	        [[5,false],[2,false],[2,false],[0,false]],
	        [[0,false],[0,false],[0,false],[0,false]]
	      ]
		})
		const firstSquare = wrapper.find('Square').at(0)
		firstSquare.simulate('click')

		const infoBoardTableTurnData = wrapper.find('InfoBoard__TableData').at(1).render().text()
		expect(infoBoardTableTurnData).toBe('1')

		const infoBoardTableAlreadyHit = wrapper.find('InfoBoard__TableData').at(9).render().text()
		expect(infoBoardTableAlreadyHit).toBe('1')

	})

	it('checks if IntoBoard gets updated correctly when we hit last ship', ()=>{
		const wrapper = mount(<App/>)
		wrapper.setState({ 
			board:[
		        [[5,false],[0,false],[0,false],[0,false]],
		        [[5,false],[0,false],[0,false],[0,false]]
		      ],
		    numberHitsLeft:2

		})
		const firstSquare = wrapper.find('Square').at(0)
		firstSquare.simulate('click')

		const fifthSquare = wrapper.find('Square').at(4)
		fifthSquare.simulate('click')

		const infoBoardTableTurnData = wrapper.find('InfoBoard__TableData').at(1).render().text()
		expect(infoBoardTableTurnData).toBe('0')

		const infoBoardTableHitData = wrapper.find('InfoBoard__TableData').at(3).render().text()
		expect(infoBoardTableHitData).toBe('0')

		const infoBoardTableMissData = wrapper.find('InfoBoard__TableData').at(5).render().text()
		expect(infoBoardTableMissData).toBe('0')

		const infoBoardTableNumberSunk = wrapper.find('InfoBoard__TableData').at(7).render().text()
		expect(infoBoardTableNumberSunk).toBe('0')

		const infoBoardTableAlreadyHit = wrapper.find('InfoBoard__TableData').at(9).render().text()
		expect(infoBoardTableAlreadyHit).toBe('0')

		const infoBoardTableNumberHitsLeft = wrapper.find('InfoBoard__TableData').at(11).render().text()
		expect(infoBoardTableNumberHitsLeft).toBe('5')

	})

	it('matches the snapshot', ()=>{
		const tree = renderer.create(<App/>).toJSON()
		expect(tree).toMatchSnapshot()
	})

});
