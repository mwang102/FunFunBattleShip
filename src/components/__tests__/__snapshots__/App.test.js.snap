// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches the snapshot 1`] = `
<div
  className="App__AppContainer-AkgWc hnJAcI"
>
  <header
    className="Header__HeaderContainer-dJGUWw egBCRU"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1
      className="App-title"
    >
      Welcome to BattleShip
    </h1>
  </header>
  <h1>
    To get started, click a square!
  </h1>
  <section
    className="App__MainSection-isHagR cDPuPv"
  >
    <table
      className="InfoBoard__TableContainer-boKKNy dlTEcW"
    >
      <tbody>
        <tr>
          <td
            className="InfoBoard__TableData-gMzDoW cllDnB"
          >
            Turns:
          </td>
          <td
            className="InfoBoard__TableData-gMzDoW cllDnB"
          >
            0
          </td>
        </tr>
        <tr>
          <td
            className="InfoBoard__TableData-gMzDoW cllDnB"
          >
            Hits:
          </td>
          <td
            className="InfoBoard__TableData-gMzDoW cllDnB"
          >
            0
          </td>
        </tr>
        <tr>
          <td
            className="InfoBoard__TableData-gMzDoW cllDnB"
          >
            Miss:
          </td>
          <td
            className="InfoBoard__TableData-gMzDoW cllDnB"
          >
            0
          </td>
        </tr>
        <tr>
          <td
            className="InfoBoard__TableData-gMzDoW cllDnB"
          >
            # Sunk:
          </td>
          <td
            className="InfoBoard__TableData-gMzDoW cllDnB"
          >
            0
          </td>
        </tr>
        <tr>
          <td
            className="InfoBoard__TableData-gMzDoW cllDnB"
          >
            Already Hit :
          </td>
          <td
            className="InfoBoard__TableData-gMzDoW cllDnB"
          >
            0
          </td>
        </tr>
        <tr>
          <td
            className="InfoBoard__TableData-gMzDoW cllDnB"
          >
            Number Hits Left :
          </td>
          <td
            className="InfoBoard__TableData-gMzDoW cllDnB"
          >
            5
          </td>
        </tr>
      </tbody>
    </table>
    <section
      className="BattleShipBoard__BoardContainer-hAezEV hOfTSv"
    >
       
      <div
        className="Row__RowContainer-kGhZeq hEhNxb"
      >
        <div
          className="Square__SquareContainer-cBVEEA gkcFPj"
          onClick={[Function]}
        />
        <div
          className="Square__SquareContainer-cBVEEA gkcFPj"
          onClick={[Function]}
        />
        <div
          className="Square__SquareContainer-cBVEEA gkcFPj"
          onClick={[Function]}
        />
        <div
          className="Square__SquareContainer-cBVEEA gkcFPj"
          onClick={[Function]}
        />
      </div>
      <div
        className="Row__RowContainer-kGhZeq egDMuO"
      >
        <div
          className="Square__SquareContainer-cBVEEA gkcFPj"
          onClick={[Function]}
        />
        <div
          className="Square__SquareContainer-cBVEEA gkcFPj"
          onClick={[Function]}
        />
        <div
          className="Square__SquareContainer-cBVEEA gkcFPj"
          onClick={[Function]}
        />
        <div
          className="Square__SquareContainer-cBVEEA gkcFPj"
          onClick={[Function]}
        />
      </div>
      <div
        className="Row__RowContainer-kGhZeq gehGJu"
      >
        <div
          className="Square__SquareContainer-cBVEEA gkcFPj"
          onClick={[Function]}
        />
        <div
          className="Square__SquareContainer-cBVEEA gkcFPj"
          onClick={[Function]}
        />
        <div
          className="Square__SquareContainer-cBVEEA gkcFPj"
          onClick={[Function]}
        />
        <div
          className="Square__SquareContainer-cBVEEA gkcFPj"
          onClick={[Function]}
        />
      </div>
      <div
        className="Row__RowContainer-kGhZeq eMrEtv"
      >
        <div
          className="Square__SquareContainer-cBVEEA gkcFPj"
          onClick={[Function]}
        />
        <div
          className="Square__SquareContainer-cBVEEA gkcFPj"
          onClick={[Function]}
        />
        <div
          className="Square__SquareContainer-cBVEEA gkcFPj"
          onClick={[Function]}
        />
        <div
          className="Square__SquareContainer-cBVEEA gkcFPj"
          onClick={[Function]}
        />
      </div>
       
    </section>
  </section>
</div>
`;
